main.py

```diff
@@ -0,0 +1,96 @@
+from typing import Dict, List
+import requests
+from dotenv import dotenv_values
+import pathlib
+
+import json
+
+
+def filter_repository_by_owner(
+    owner_name: str,
+    repositories: List[Dict],
+    apply_filter: bool = False,
+) -> List[Dict]:
+    """
+    Receive repositories and filter them by owner_name. Return the subset of repositories filtered.
+    If apply_filter = False, return repositories as is.
+    """
+    pass
+
+
+def get_metadata(token: str, path_dir: pathlib.Path, pages: int = 15) -> list:
+    session = requests.Session()
+    metadata = []
+    pages = 15
+    max_per_page = 100  # Max value accepted by github api
+
+    my_headers = {"Authorization": f"Bearer {token}"}
+
+    for page in range(1, pages):
+        params = {"per_page": max_per_page, "page": page}
+        response = session.get(
+            "https://api.github.com/user/repos", headers=my_headers, params=params
+        )
+        resp = response.json()
```

Comment:
Change the name of this variable to `response_json`, I dont know what `resp` means neither the difference between it and the variable `response`. We need to give good names to the variables to keep the code readable.
Please watch videos #1, #2 and #3 from this series: https://www.youtube.com/watch?v=9w3o9NHXqu0&ab_channel=FilipeDeschamps